src/services/loopd/loop-service.ts

```diff
@@ -0,0 +1,196 @@
+// https://lightning.engineering/loopapi/index.html#service-swapclient
+import util from "util"
+
+import * as grpc from "@grpc/grpc-js"
+import { SwapClientNotResponding, SwapServiceError } from "@domain/swap/errors"
+import { SwapState as SwapStateType } from "@domain/swap/index"
+import { SwapType as DomainSwapType } from "@domain/swap"
+import { ServiceClient } from "@grpc/grpc-js/build/src/make-client"
+import { wrapAsyncFunctionsToRunInSpan } from "@services/tracing"
+import { BtcNetwork } from "@domain/bitcoin"
+
+import { SwapClientClient } from "./protos/loop_grpc_pb"
+import {
+  FailureReason,
+  QuoteRequest,
+  OutQuoteResponse,
+  LoopOutRequest,
+  SwapResponse,
+  MonitorRequest,
+  SwapState,
+  SwapType,
+  SwapStatus,
+} from "./protos/loop_pb"
+
+export const LoopService = ({
+  macaroon,
+  tlsCert,
+  grpcEndpoint,
+  btcNetwork,
+}: LoopdConfig): ISwapService => {
+  const mac = Buffer.from(macaroon, "base64").toString("hex")
+  const tls = Buffer.from(tlsCert, "base64")
+  const swapClient: ServiceClient = createClient(mac, tls, grpcEndpoint)
+
+  const clientHealthCheck = util.promisify<QuoteRequest, OutQuoteResponse>(
+    swapClient.loopOutQuote.bind(swapClient),
+  )
+  const clientSwapOut = util.promisify<LoopOutRequest, SwapResponse>(
+    swapClient.loopOut.bind(swapClient),
+  )
+
+  const healthCheck = async (): Promise<boolean> => {
+    try {
+      const request = new QuoteRequest()
+      request.setAmt(500000)
+      const resp = await clientHealthCheck(request)
+      const fee = resp.getSwapFeeSat()
```

Comment:
does this result in a request to the LL server? 

maybe in that a simpler request that doesn't require a query to a server outside our cluster would be preferable for a healthcheck?

Reply:
yes, i updated the health check to use a simpler call to an internal token request service